cmake/helpers/configure.cmake

```
@@ -44,6 +44,21 @@ check_type_size("long int" SIZEOF_LONG_INT)
 check_type_size("void*" SIZEOF_VOIDP)
 check_type_size("size_t" SIZEOF_SIZE_T)
 
+# Check whether the type `_Float16` exists, and whether type
+# conversions work (there might be linker problems)
+check_cxx_source_compiles(
+    "
+      #ifdef GDAL_DISABLE_FLOAT16
+        Explicitly disable _Float16 support
+      #endif
+      int main() {
+        _Float16 h = 1;
+        float f = h;
+        double d = h;
```

to potentially avoid warning about unused variables in the most pedantic warning levels
```suggestion
        double d = h;
        return d != 0 ? 0 : 1;
```

---

Done. Thanks.